backend/test/integration/langchain.test.ts

```diff
@@ -209,11 +210,9 @@ describe('langchain integration tests ', () => {
 	});
 
 	test('GIVEN the prompt evaluation model is initialised WHEN it is asked to evaluate an input AND it does not respond in the correct format THEN it returns a final decision of false', async () => {
-		mockFromLLM.mockImplementation(() => mockPromptEvalChain);
```

Comment:
So, fromLLM is basically a fancy static method within the `RetrievalQAChain` class that constructs an instance of `RetrievalQAChain`. Therefore every time we mock it to return the `mockPromptEvalChain`, it was wrong.

Further, since it should actually return `mockRetrievalQAChain` for every test, we don't need to mock differently it per test, so I've moved the mockImplementation to the beforeEach